aws/resource_aws_xray_sampling_rule.go

```diff
@@ -0,0 +1,212 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/xray"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsXraySamplingRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsXraySamplingRuleCreate,
+		Read:   resourceAwsXraySamplingRuleRead,
+		Update: resourceAwsXraySamplingRuleUpdate,
+		Delete: resourceAwsXraySamplingRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"rule_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"resource_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"priority": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"fixed_rate": {
+				Type:     schema.TypeFloat,
+				Required: true,
+			},
+			"reservoir_size": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"service_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"host": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"http_method": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"url_path": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"version": {
```

Comment:
Two notes:
* Rather than return an error during update, this attribute should use [`ForceNew`](https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#forcenew) to trigger Terraform resource recreation
* We can add validation for this attribute to match the [API reference](https://docs.aws.amazon.com/xray/latest/api/API_SamplingRule.html) via:

```go
ForceNew: true,
ValidateFunc: validation.IntAtLeast(1),
```